Updates to rcutils to make rosdoc2 generation happier.

clalancette opened the pull request:
In particular:

* Fix a mostly duplicated and slightly wrong formatted docblock at the top of sha256.c
* Add in a few field names so doxygen can resolve them.
* Make a doxygen-only function signature of sha256_final, since rosdoc2 doesn't understand uint8_t[]
* Modify the configuration to include all needed files.

With this in place, all but one warning are removed when generating the documentation.

clalancette commented on include/rcutils/sha256.h:
```diff
@@ -83,10 +79,18 @@ void rcutils_sha256_update(rcutils_sha256_ctx_t * ctx, const uint8_t * data, siz
  * \return void
  */
+#ifdef DOXYGEN_ONLY
+// rosdoc2 doesn't understand uint8_t[], so make it a pointer for documentation purposes
```
> Is it that rosdoc2 doesn't understand it or is it doxygen?

I'm not 100% sure, but I actually think it is either exhale or breathe.  That is, the generated `.rst` files that come out of doxygen/breathe/exhale pipeline have the function signature as:

```
void rcutils_sha256_final(rcutils_sha256_ctx_t * ctx, uint8_t);
```

wjwwood replied:
Oh ok, well maybe word it like that? When I read this comment it doesn't make sense to me. Maybe even just "one of the tools used by rosdoc2 misinterprets this as `uint8_t` so make it a pointer for documentation purposes"?

clalancette replied:
Fixed in 5b3a510.

clalancette commented:
New CI with the latest changes (only checking up to rcutils, since the change since the last CI was just to a comment):
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=18287)](http://ci.ros2.org/job/ci_linux/18287/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=12813)](http://ci.ros2.org/job/ci_linux-aarch64/12813/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=18954)](http://ci.ros2.org/job/ci_windows/18954/)

wjwwood reviewed (approved):
Lgtm, thanks for updating it
